app/controllers/api/physical_servers_controller.rb

```diff
@@ -41,7 +41,8 @@ def restart_mgmt_controller_resource(type, id, _data)
     end
 
     def refresh_resource(type, id, _data = nil)
-      raise BadRequestError, "Must specify an id for refreshing a #{type} resource" unless id
```

Comment:
I think it would be OK not to extract this into a private method, since that alters the message to talk about "changing" here, instead of "refreshing". Again, I think it's an acceptable structural similarity.